### PR TITLE
feat: SQSCANGHA-6 add ability to pass certificate to action

### LINKED ISSUE #26: Fails to connect to Sonarqube server signed by corporate PKI
Hi,

We've an internal sonarqube server, that is using a certificate signed by corporate PKI.
Action doesn't want to connect to it.

> org.sonarsource.scanner.api.internal.ScannerException: Unable to execute SonarScanner analysis
> Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

I had to fork the action repo and add following lines to the Dockerfile
```lang=Dockerfile
# Add certificate in java keystore
COPY sonarqube.corp.net.pem .
RUN keytool -keystore /etc/ssl/certs/java/cacerts -storepass changeit -noprompt -trustcacerts -importcert -alias sonarqube -file sonarqube.corp.net.pem
```

There should be an option to deal with the issue (use cert store of the host, accept as arg the hash of the certificate to trust...). It may be possible to solve that with java arg, but if it's the case it may be great to add it in the Readme.

Thanks

### PR DESCRIPTION
**Pull request by Aeolun**
Since the solution to this problem seemed kind of straightforward to me, and I was having this same issue, here is a fix. This should make it easier for any other person with this problem to  add certificates through an environment variable going forward.

Fixes #26

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -13,6 +13,12 @@ if [[ -z "${SONAR_HOST_URL}" ]]; then
   exit 1
 fi
 
+if [[ -n "${SONAR_ROOT_CERT}" ]]; then
+  echo "Adding custom root certificate to java certificate store"
+  echo "${SONAR_ROOT_CERT}" > /tmp/tmpcert.pem
+  keytool -keystore /etc/ssl/certs/java/cacerts -storepass changeit -noprompt -trustcacerts -importcert -alias sonarqube -file /tmp/tmpcert.pem
```
Comment (dbmeneses) [inline entrypoint.sh:19]: I'd add `rm /tmp/tmpcert.pem`

File: README.md [hunk 2]
```diff
@@ -91,6 +102,7 @@ More information about possible analysis parameters can be found in [the documen
 
 - `SONAR_TOKEN` – **Required** this is the token used to authenticate access to SonarQube. You can read more about security tokens [here](https://docs.sonarqube.org/latest/user-guide/user-token/). You can set the `SONAR_TOKEN` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
 - `SONAR_HOST_URL` – **Required** this tells the scanner where SonarQube is hosted. You can set the `SONAR_HOST_URL` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
+- `SONAR_ROOT_CERT` – This is used to pass the scanner a custom root certificate (in PEM format), that should be used for validating the sonarqube server certificate. You can set the `SONAR_ROOT_CERT` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
```
Comment (dbmeneses) [inline README.md:105]:
```suggestion
- `SONAR_ROOT_CERT` – Holds an additional root certificate (in PEM format) that is used to validate the SonarQube server certificate. You can set the `SONAR_ROOT_CERT` environment variable in the "Secrets" settings page of your repository, or you can add them at the level of your GitHub organization (recommended).
```

**Review by dbmeneses — APPROVED** [review]
Minor comments otherwise LGTM. @wouter-admiraal-sonarsource could you please review it as well?

File: README.md [hunk 3]
```diff
@@ -63,6 +63,17 @@ If your source code file names contain special characters that are not covered b
         LC_ALL: "ru_RU.UTF-8"
 ```
 
+If your sonarqube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
```
Comment (wouter-admiraal-sonarsource) [inline README.md:66]: Nitpicking:

```suggestion
If your SonarQube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
```

File: README.md [hunk 4]
```diff
@@ -63,6 +63,17 @@ If your source code file names contain special characters that are not covered b
...
+
+```yaml
+    - name: SonarQube Scan
+      uses: sonarsource/sonarqube-scan-action@master
+      env:
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        SONAR_ROOT_CERT: %{{ secrets.SONAR_ROOT_CERT }}
```
Comment (wouter-admiraal-sonarsource) [inline README.md:74]:
```suggestion
        SONAR_ROOT_CERT: ${{ secrets.SONAR_ROOT_CERT }}
```

**Review by wouter-admiraal-sonarsource — CHANGES_REQUESTED** [review]
There is one change need in the example, otherwise LGTM.

**Review by ilostmyrepo — COMMENTED** [review]
accept

**Comment by dbmeneses** [comment]
@Aeolun when you get a chance please address the feedback so that we can merge this PR

**Comment by marcocrasso** [comment]
Is there any ETA for this to get merged?

File: README.md [hunk 5]
```diff
@@ -63,6 +63,17 @@ If your source code file names contain special characters that are not covered b
         LC_ALL: "ru_RU.UTF-8"
 ```
 
+If your SonarQube server users a self-signed certificate, you can pass a root certificate (in PEM format) to use for validation to the java certificate store:
```
Comment (wouter-admiraal-sonarsource) [inline README.md:66]:
```suggestion
If your SonarQube server uses a self-signed certificate, you can pass a root certificate (in PEM format) to the java certificate store:
```

File: entrypoint.sh [hunk 6]
```diff
@@ -13,6 +13,12 @@ if [[ -z "${SONAR_HOST_URL}" ]]; then
   exit 1
 fi
 
+if [[ -n "${SONAR_ROOT_CERT}" ]]; then
+  echo "Adding custom root certificate to java certificate store"
+  echo "${SONAR_ROOT_CERT}" > /tmp/tmpcert.pem
```
Comment (wouter-admiraal-sonarsource) [inline entrypoint.sh:18]:
```suggestion
  rm -f /tmp/tmpcert.pem
  echo "${SONAR_ROOT_CERT}" > /tmp/tmpcert.pem
```

**Review by wouter-admiraal-sonarsource — APPROVED** [review]

**Pull request merged**